Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -211,7 +210,7 @@ actor ScopeGenerator: CustomStringConvertible {
                             .unwrappedTypeDescription
                             .asInstantiatedType
                             .asSource
-                        let propertyDeclaration = if typeDescription == unwrappedTypeDescription {
+                        let propertyDeclaration = if !instantiable.declarationType.isExtension, typeDescription == unwrappedTypeDescription {
```

Comment:
this check and the check below are necessary to ensure our generated code is not ambiguous